numpy/lib/tests/test_function_base.py

```diff
@@ -3331,6 +3332,31 @@ def test_keepdims(self):
         assert_equal(np.percentile(d, [1, 7], axis=(0, 3),
                                    keepdims=True).shape, (2, 1, 5, 7, 1))
 
+    @pytest.mark.parametrize('q', [7, [1, 7]])
+    @pytest.mark.parametrize(
+        argnames='axis',
+        argvalues=[
+            None,
+            1,
+            (1,),
+            (0, 1),
+            (~2, ~0),
+        ]
+    )
+    def test_keepdims_out(self, q, axis):
+        d = np.ones((3, 5, 7, 11))
+        if axis is None:
+            shape_out = (1,) * d.ndim
+        else:
+            axis_norm = normalize_axis_tuple(axis, d.ndim)
+            shape_out = tuple(
+                1 if i in axis_norm else d.shape[i] for i in range(d.ndim))
+        shape_out = np.shape(q) + shape_out
+
+        out = np.empty(shape_out)
+        result = np.percentile(d, q, axis=axis, keepdims=True, out=out)
+        assert_equal(result.shape, shape_out)
```

Comment:
Good thing you had me add this, because now the tests are failing. I guess it's because the indexing op created a view right? I need to think about what the best way to get rid of the view is.

Reply:
Ah OK... if there is stuff like that going on that has been there, maybe just check `np.may_share_memory()`, thanks!

Reply:
Hmm, it should really be `out` that is returned -- the reshape is not needed given the view that was taken.  But that means one may indeed have to move this whole `keepdims` stanza inside `_ureduce` after all.

Reply:
I agree. I think it breaks the principle of least surprise having it return a view of `out`. I'll try sucking this `keepdims` stanza inside `_ureduce`.